toolkit/primitives/domain/src/lib.rs

```diff
@@ -298,15 +311,37 @@ impl MainchainKeyHash {
 	pub fn from_vkey(vkey: [u8; 32]) -> Self {
 		Self(blake2b(&vkey))
 	}
+
+	pub fn from_bech32_address(address: &str) -> Result<Self, String> {
```

Comment:
bech32 is a strong based type, maybe instead of taking a &str, would be good to implement TryFrom<bech32> instead